drivers/azure/azureutil/auth.go

```diff
@@ -33,11 +29,25 @@ var (
 // will automatically refresh the specified token and will call the refresh
 // callback function we implement here. This way we will always be storing a
 // token with a refresh_token saved on the machine.
+//
+// 2. Azure Service Principal Account
+//
+// This is designed for headless authentication to Azure APIs but requires more
+// steps from user to create a Service Principal Account and provide its
+// credentials to the machine driver.
+
+var (
+	// AD app id for docker-machine driver in various Azure realms
+	appIDs = map[string]string{
+		azure.PublicCloud.Name: "637ddaba-219b-43b8-bf19-8cea500cf273",
```

Comment:
Don't know about all the realms, but does this have to be populated for each Realm or there are only two?